File: frontend/types/menu_item.ts
export interface MenuItem {
    menu_item_id: number;
    menu_id: number;
    name: string;
    description: string | null;
    price: number;
    is_available: boolean;
}

export interface MenuItemCreate {
    name: string;
    description?: string | null;
    price: number;
    is_available?: boolean;
}

export interface MenuItemUpdate {
    name?: string;
    description?: string | null;
    price?: number;
    is_available?: boolean;
}

File: backend/app/schemas/menu_item.py
from decimal import Decimal

from pydantic import BaseModel, ConfigDict


class MenuItemCreate(BaseModel):
    name: str
    description: str | None = None
    price: Decimal
    is_available: bool = True


class MenuItemUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    price: Decimal | None = None
    is_available: bool | None = None


class MenuItemResponse(BaseModel):
    menu_item_id: int
    menu_id: int
    name: str
    description: str | None = None
    price: Decimal
    is_available: bool

    model_config = ConfigDict(from_attributes=True)